pex/common.py

```diff
@@ -464,10 +464,25 @@ def unlock():
             os.path.join(head, ".{}.atomic_directory.lck".format(tail or "here")),
             os.O_CREAT | os.O_WRONLY,
         )
-        # N.B.: Since lockf operates on an open file descriptor and these are guaranteed to be
-        # closed by the operating system when the owning process exits, this lock is immune to
-        # staleness.
-        fcntl.lockf(lock_fd, fcntl.LOCK_EX)  # A blocking write lock.
+        while True:
```

Comment:
Hmm should we cap the retry/timeout here?

Reply:
I'm actually thinking we shouldn't. We can't know how long this would take (maybe we're downloading `pytorch`, which is several GB) and the blocking write itself doesn't have a timeout, meaning we haven't really changed behavior all that much.